### DISCUSSION TITLE
How to create DrawingContext from bitmap or writeablebitmap?

### DISCUSSION THREAD

**Discussion by sandsc** [AvaloniaUI/Avalonia#11493] [Q&A]
A graphics object can be created from Image in GDI Graphics.FromImage， then we can draw on this image。how to do in avalonia?

**Comment by maxkatz6** [comment]
Avalonia isn't an image processing library like GDI, so its API is more limited in this aspect. You can use crossplatform solutions like SkiaSharp or ImageSharp. 

In Avalonia, though, you can use RenderTargetBitmap, copy/draw your original bitmap there and add modifications on top of it.

**Reply by Gillibald** [reply]
RenderTargetBitmap uses a WriteableBitmap under the hood so it will give you want you want
